presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -282,6 +284,18 @@ private void chooseDistribution(PartitioningHandle distribution)
                     partitioningHandle);
         }
 
+        private boolean isCompatibleSystemPartitioning(PartitioningHandle distribution)
+        {
+            ConnectorPartitioningHandle currentHandle = partitioningHandle.get().getConnectorHandle();
+            ConnectorPartitioningHandle distributionHandle = distribution.getConnectorHandle();
+            if ((currentHandle instanceof SystemPartitioningHandle) &&
+                    (distributionHandle instanceof SystemPartitioningHandle)) {
+                return ((SystemPartitioningHandle) currentHandle).getPartitioning() ==
```

Comment:
My concern is that something might be incorrectly tagged as FIXED-HASH when it's actually not guaranteed to be hash-distributed. It may not matter now, but it could be a ticking time bomb.

Another thing I'm looking into is whether this is a symptom of the plan being incorrect in the first place. If you've already done this analysis, please let me know what you've found.